studio/proxy-server.mjs

```diff
@@ -11,25 +13,40 @@ const corsHeaders = {
 };
 
 const targetOrigin = process.argv[2];
+const keyPath = process.argv[3];
+const certPath = process.argv[4];
 
 if (!targetOrigin) {
-    console.error('Usage: node proxy.mjs <targetOrigin>');
+    console.error('Usage: node proxy.mjs <targetOrigin> [keyPath] [certPath]');
     process.exit(1);
 }
 
 const targetUrl = new URL(targetOrigin);
 
-const server = createServer((req, res) => {
-    const { method, headers, url } = req;
+// Variables to hold the server and protocol
+let server;
+let serverProtocol;
+let serverPort;
+
+if (keyPath && certPath) {
+    // Read the SSL certificate and private key
+    const httpsOptions = {
+        key: readFileSync(keyPath),
+        cert: readFileSync(certPath),
+    };
+
+    server = createHttpsServer(httpsOptions, requestHandler);
+    serverProtocol = 'https';
+    serverPort = 8443; // Use 8443 for HTTPS
+} else {
```

Comment:
why leaving http option? i would remove it tbh

Reply:
it is needed on local. If we drop local proxy support we can remove it.